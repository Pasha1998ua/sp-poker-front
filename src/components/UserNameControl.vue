<template>
  <div class="set-user-name-block">
    <button @click="setUserName">Set User Name</button>
    <input v-model="inputVal">
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import localStorageHelper from '@/helpers/localStorageHelper'

export default defineComponent({
  name: 'MainPage',
  data () {
    return {
      inputVal: '' as string
    }
  },
  methods: {
    setUserName () {
      localStorageHelper.setUserName(this.inputVal)
    }
  }
})
</script>

<style scoped lang="scss">
</style>
