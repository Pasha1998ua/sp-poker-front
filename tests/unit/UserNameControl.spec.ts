import { shallowMount } from '@vue/test-utils'
import UserNameControl from '@/components/UserNameControl.vue'

describe('UserNameControl.vue', () => {
  it('renders page correctly', () => {
    const wrapper = shallowMount(UserNameControl)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
