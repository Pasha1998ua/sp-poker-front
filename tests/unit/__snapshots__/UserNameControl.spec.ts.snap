// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNameControl.vue renders page correctly 1`] = `<div class="set-user-name-block"><button>Set User Name</button><input></div>`;
